libs/ledger-live-common/src/api/socket.ts

```diff
@@ -91,45 +89,61 @@ export const createDeviceSocket = (
               nonce,
               apdu,
             });
-            // specific logic for detecting allow manager
-            let allowManagerAwaitingUser = false;
+
+            // Detect the specific exchange that triggers the allow secure channel request.
+            let pendingUserAllowSecureChannel = false;
 
             if (apdu.slice(0, 2).toString("hex") === "e051") {
-              timeoutForAllowManager = setTimeout(() => {
+              pendingUserAllowSecureChannel = true;
+              allowSecureChannelTimeout = setTimeout(() => {
                 if (unsubscribed) return;
-                allowManagerAwaitingUser = true;
                 o.next({
                   type: "device-permission-requested",
-                  wording: "Ledger Manager",
+                  wording: "Ledger Manager", // TODO make this dynamic per fw version to match device
                 });
-              }, ALLOW_MANAGER_DELAY);
+                // Nb Permission is only requested once per reboot, delaying the event
+                // prevents the UI from flashing the rendering for allowing.
+              }, ALLOW_SECURE_CHANNEL_DELAY);
             }
 
             const r = await transport.exchange(apdu);
 
-            if (timeoutForAllowManager) {
-              // if the exchange is faster than the timeout, we pass through because it's not "awaiting" user action
-              clearTimeout(timeoutForAllowManager as number);
-              timeoutForAllowManager = null;
+            if (allowSecureChannelTimeout) {
+              clearTimeout(allowSecureChannelTimeout as number);
```

Comment:
You can define: `let allowSecureChannelTimeout: NodeJS.Timeout | null = null;` withouth `number` and then only have:

```TS
clearTimeout(allowSecureChannelTimeout);
```

IDK why TS on VSCode wanted `clearTimeout` arg as an input, it was the same on my side. But like above TS should work.

Reply:
uhmm I didn't write this, but sure, your suggestion works